src/adagrams.js

```diff
@@ -1,8 +1,124 @@
 const Adagrams = {
   drawLetters() {
     // Implement this method for wave 1
+    let quantityByLetter = {
+      A: 9,
+      B: 2,
+      C: 2,
+      D: 4,
+      E: 12,
+      F: 2,
+      G: 3,
+      H: 2,
+      I: 9,
+      J: 1,
+      K: 1,
+      L: 4,
+      M: 2,
+      N: 6,
+      O: 8,
+      P: 2,
+      Q: 1,
+      R: 6,
+      S: 4,
+      T: 6,
+      U: 4,
+      V: 2,
+      W: 2,
+      X: 1,
+      Y: 2,
+      Z: 1
+    };
+
+    let letters = "";
+    Object.keys(quantityByLetter).forEach(key => {
+      let lettersQuant = key.repeat([quantityByLetter[key]]);
+      letters += lettersQuant;
+    });
+    letters = letters.split("");
+
+    let drawn = [];
+    let i = 0;
+    while (i < 10) {
+      drawn.push(letters[Math.floor(Math.random() * letters.length)]);
+      i++;
+    }
+
+    return drawn;
+  },
+
+  usesAvailableLetters(input, lettersInHand) {
+    let word = input.split("");
+    let handLetters = {};
+    lettersInHand.forEach(letter => {
+      if (handLetters.hasOwnProperty(letter)) {
+        handLetters[letter] += 1;
+      } else {
+        handLetters[letter] = 1;
+      }
+    });
+
+    let wordLetters = {};
+    word.forEach(letter => {
+      if (wordLetters.hasOwnProperty(letter)) {
+        wordLetters[letter] += 1;
+      } else {
+        wordLetters[letter] = 1;
+      }
+    });
+
+    for (const letter of word) {
+      if (handLetters[letter] === undefined || wordLetters[letter] > handLetters[letter]) {
+        return false;
+      }
+    }
+    return true;
+  },
+
+  scoreWord(word) {
+  const valuesLetter = {
```

Comment:
Style note: move this object outside the function `scoreWord()` but inside the module `Adagrams`